src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,32 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        /// <summary>
+        /// Provides an entry point to append code (after execute content) to a generator razor class.
+        /// </summary>
+        /// <param name="writer">writer writes the additional content</param>
+        /// <param name="chunks">a list of chunks representing the generated page</param>
```

Comment:
Sentences with an initial cap and a final period please. Also suggest `The <see cref="CSharpCodeWriter"/> to receive the additional content.` and `The list of <see cref="Chunk"/>s for the generated program.`

Similar comments for the `DecorateChunks()` parameters.